Gracefully handle `AnnotationInstance` being null in `InjectionPointsProcessor.syntheticBeans`

Some potential edge cases can result in an NPE that we should try to avoid. And at a minimum throw a friendlier exception that could perhaps hint at what the problem is.

E.g something like this results in an NPE during the build phase:

```
@Produce
@Inject
FluentProducerTemplate template;
```

The following stack trace might help:
```
Caused by: java.lang.NullPointerException
    at org.apache.camel.quarkus.core.deployment.InjectionPointsProcessor.produceBeans (InjectionPointsProcessor.java:295)
    at org.apache.camel.quarkus.core.deployment.InjectionPointsProcessor.syntheticBeans (InjectionPointsProcessor.java:253)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at io.quarkus.deployment.ExtensionLoader$2.execute (ExtensionLoader.java:820)
    at io.quarkus.builder.BuildContext.run (BuildContext.java:277)
    at org.jboss.threads.ContextHandler$1.runWith (ContextHandler.java:18)
    at org.jboss.threads.EnhancedQueueExecutor$Task.run (EnhancedQueueExecutor.java:2449)
    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run (EnhancedQueueExecutor.java:1478)
    at java.lang.Thread.run (Thread.java:829)
    at org.jboss.threads.JBossThread.run (JBossThread.java:501)
```

--- a/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/InjectionPointsProcessor.java
+++ b/extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/InjectionPointsProcessor.java
@@ -62,6 +62,7 @@ import org.jboss.jandex.AnnotationTarget.Kind;
 import org.jboss.jandex.ClassInfo;
 import org.jboss.jandex.DotName;
 import org.jboss.jandex.FieldInfo;
+import org.jboss.jandex.IndexView;
 import org.jboss.jandex.MethodInfo;
 import org.jboss.jandex.Type;
 import org.jboss.logging.Logger;
@@ -230,7 +231,7 @@ public class InjectionPointsProcessor {
             switch (target.kind()) {
             case FIELD: {
                 final FieldInfo field = target.asField();
-                endpointInjectBeans(recorder, syntheticBeans, annot, field.type().name());
+                endpointInjectBeans(recorder, syntheticBeans, index.getIndex(), annot, field.type().name());
                 break;
             }
             case METHOD: {
@@ -250,8 +251,8 @@ public class InjectionPointsProcessor {
             switch (target.kind()) {
             case FIELD: {
                 final FieldInfo field = target.asField();
-                produceBeans(recorder, capabilities, syntheticBeans, proxyDefinitions, beanCapabilityAvailable, annot,
-                        field.type().name(), field.name(), field.declaringClass().name());
+                produceBeans(recorder, capabilities, syntheticBeans, proxyDefinitions, beanCapabilityAvailable,
+                        index.getIndex(), annot, field.type().name(), field.name(), field.declaringClass().name());
                 break;
             }
             case METHOD: {
@@ -269,6 +270,7 @@ public class InjectionPointsProcessor {
             BuildProducer<SyntheticBeanBuildItem> syntheticBeans,
             BuildProducer<NativeImageProxyDefinitionBuildItem> proxyDefinitions,
             AtomicReference<Boolean> beanCapabilityAvailable,
+            IndexView index,
             AnnotationInstance annot, final DotName fieldType, String annotationTarget, DotName declaringClass) {
         try {
             Class<?> clazz = Class.forName(fieldType.toString(), false,
@@ -279,7 +281,7 @@ public class InjectionPointsProcessor {
                                 .configure(fieldType)
                                 .setRuntimeInit().scope(Singleton.class)
                                 .supplier(
-                                        recorder.createProducerTemplate(annot.value().asString()))
+                                        recorder.createProducerTemplate(annot.valueWithDefault(index).asString()))
                                 .addQualifier(annot)
                                 .done());
                 /*
@@ -292,7 +294,7 @@ public class InjectionPointsProcessor {
                                 .configure(fieldType)
                                 .setRuntimeInit().scope(Singleton.class)
                                 .supplier(
-                                        recorder.createFluentProducerTemplate(annot.value().asString()))
+                                        recorder.createFluentProducerTemplate(annot.valueWithDefault(index).asString()))
                                 .addQualifier(annot)
                                 .done());
                 /*
@@ -320,7 +322,7 @@ public class InjectionPointsProcessor {
                                 .configure(fieldType)
                                 .setRuntimeInit().scope(Singleton.class)
                                 .supplier(
-                                        recorder.produceProxy(clazz, annot.value().asString()))
+                                        recorder.produceProxy(clazz, annot.valueWithDefault(index).asString()))
                                 .addQualifier(annot)
                                 .done());
             }
@@ -330,7 +332,7 @@ public class InjectionPointsProcessor {
     }
 
     private void endpointInjectBeans(CamelRecorder recorder, BuildProducer<SyntheticBeanBuildItem> syntheticBeans,
-            AnnotationInstance annot, final DotName fieldType) {
+            IndexView index, AnnotationInstance annot, final DotName fieldType) {
         try {
             Class<?> clazz = Class.forName(fieldType.toString());
             if (Endpoint.class.isAssignableFrom(clazz)) {
@@ -339,7 +341,7 @@ public class InjectionPointsProcessor {
                                 .configure(fieldType)
                                 .setRuntimeInit().scope(Singleton.class)
                                 .supplier(
-                                        recorder.createEndpoint(annot.value().asString(),
+                                        recorder.createEndpoint(annot.valueWithDefault(index).asString(),
                                                 (Class<? extends Endpoint>) clazz))
                                 .addQualifier(annot)
                                 .done());
@@ -349,7 +351,7 @@ public class InjectionPointsProcessor {
                                 .configure(fieldType)
                                 .setRuntimeInit().scope(Singleton.class)
                                 .supplier(
-                                        recorder.createProducerTemplate(annot.value().asString()))
+                                        recorder.createProducerTemplate(annot.valueWithDefault(index).asString()))
                                 .addQualifier(annot)
                                 .done());
                 /*
@@ -362,7 +364,7 @@ public class InjectionPointsProcessor {
                                 .configure(fieldType)
                                 .setRuntimeInit().scope(Singleton.class)
                                 .supplier(
-                                        recorder.createFluentProducerTemplate(annot.value().asString()))
+                                        recorder.createFluentProducerTemplate(annot.valueWithDefault(index).asString()))
                                 .addQualifier(annot)
                                 .done());
                 /*

